contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,415 @@
+
+        AvroToPojo.ActiveFieldInfo afi = columnFieldSetters.get(i);
+        afi = columnFieldSetters.get(i);
+        SupportType st = afi.fieldInfo.getType();
+        Object val = null;
+
+        try {
+          val = tuple.get(afi.fieldInfo.getColumnName());
+        } catch (Exception e) {
+          LOG.error("Could not find field -" + afi.fieldInfo.getColumnName() + "- in the generic record");
+          val = null;
+          fieldErrorCnt++;
+        }
+
+        if (val == null) {
+          continue;
+        }
+
+        try {
+          switch (st) {
+            case BOOLEAN:
+              ((PojoUtils.SetterBoolean<Object>)afi.setterOrGetter).set(newObj,
+                  (boolean)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case DOUBLE:
+              ((PojoUtils.SetterDouble<Object>)afi.setterOrGetter).set(newObj,
+                  (double)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case FLOAT:
+              ((PojoUtils.SetterFloat<Object>)afi.setterOrGetter).set(newObj,
+                  (float)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case INTEGER:
+              ((PojoUtils.SetterInt<Object>)afi.setterOrGetter).set(newObj,
+                  (int)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case STRING:
+              ((PojoUtils.Setter<Object, String>)afi.setterOrGetter).set(newObj,
+                  new String(tuple.get(afi.fieldInfo.getColumnName()).toString()));
+              break;
+
+            case LONG:
+              ((PojoUtils.SetterLong<Object>)afi.setterOrGetter).set(newObj,
+                  (long)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            default:
+              ((PojoUtils.Setter<Object, byte[]>)afi.setterOrGetter).set(newObj,
+                  serialize(tuple.get(afi.fieldInfo.getColumnName())));
+              break;
+          }
+        } catch (AvroRuntimeException | IOException e) {
+          LOG.error("Exception in setting value" + e.getMessage());
+          fieldErrorCnt++;
+        }
+
+      }
+    } catch (Exception ex) {
+      LOG.error("Generic Exception in setting value" + ex.getMessage());
+      errorCnt++;
+      newObj = null;
+    }
+    return newObj;
+  }
+
+  /**
+   * Use reflection to generate field info values if the user has not provided
+   * the inputs mapping
+   * 
+   * @return String representing the POJO field to Avro field mapping
+   */
+  public String generateFieldInfoInputs(Class<?> cls)
+  {
+    java.lang.reflect.Field[] fields = cls.getDeclaredFields();
+    StringBuilder sb = new StringBuilder();
+
+    for (int i = 0; i < fields.length; i++) {
+      java.lang.reflect.Field f = fields[i];
+      Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+      sb.append(f.getName() + FIELD_SEPARATOR + f.getName() + FIELD_SEPARATOR + c.getSimpleName().toUpperCase()
```

Comment:
This is very picky... But you already are working with StringBuilder.
you can just do sb.append(...).append(...)...